controllers/flowcollector_controller.go

```diff
@@ -137,14 +137,22 @@ func (r *FlowCollectorReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	// OVS config map for CNO
-	ovsConfigController := ovs.NewFlowsConfigController(clientHelper,
-		ns,
-		desired.Spec.ClusterNetworkOperator.Namespace,
-		ovsFlowsConfigMapName,
-		r.lookupIP)
-	if err := ovsConfigController.Reconcile(ctx, desired, gfReconciler.GetServiceName(desired.Spec.Kafka)); err != nil {
-		return ctrl.Result{},
-			fmt.Errorf("failed to reconcile ovs-flows-config ConfigMap: %w", err)
+	if desired.Spec.ClusterNetworkOperator.Namespace != "" {
```

Comment:
maybe I should replace the namespace check with something more explicit like a new field `Spec.ClusterNetworkOperator.Enabled` to switch between ovn-k and CNO

Reply:
So I'm using the discovery client to check for presence of the CNO now.
I haven't done the ebpf switch as you suggest although I agree it will be nice